idaes/models_extra/gas_solid_contactors/unit_models/fixed_bed_1D.py

```diff
@@ -0,0 +1,2047 @@
+                self,
+                wrt=self.length_domain,
+                nfe=self.config.finite_elements,
+                scheme=self.config.transformation_scheme,
+            )
+        elif self.config.transformation_method == "dae.collocation":
+            self.discretizer = TransformationFactory(self.config.transformation_method)
+            self.discretizer.apply_to(
+                self,
+                wrt=self.length_domain,
+                nfe=self.config.finite_elements,
+                ncp=self.config.collocation_points,
+                scheme=self.config.transformation_scheme,
+            )
+
+    def _make_performance(self):
+        """
+        Constraints for unit model.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        # local aliases used to shorten object names
+        gas_phase = self.config.gas_phase_config
+        solid_phase = self.config.solid_phase_config
+
+        # Get units meta data from property packages
+        units_meta_gas = gas_phase.property_package.get_metadata().get_derived_units
+        units_meta_solid = solid_phase.property_package.get_metadata().get_derived_units
+
+        # Declare Mutable Parameters
+        self.eps = Param(
+            mutable=True,
+            default=1e-8,
+            doc="Smoothing Factor for Smooth IF Statements",
+            units=pyunits.dimensionless,
+        )
+
+        # Unit Model variables
+        self.bed_diameter = Var(
+            initialize=1, doc="Reactor diameter", units=units_meta_solid("length")
+        )
+        self.bed_area = Var(
+            domain=Reals,
+            initialize=1,
+            doc="Reactor cross-sectional area",
+            units=units_meta_solid("area"),
+        )
+
+        # Phase specific variables
+        self.velocity_superficial_gas = Var(
+            self.flowsheet().time,
+            self.length_domain,
+            domain=Reals,
+            initialize=0.05,
+            doc="Gas superficial velocity",
+            units=units_meta_gas("velocity"),
+        )
+
+        # Dimensionless numbers, mass and heat transfer coefficients
+        self.Re_particle = Var(
+            self.flowsheet().time,
+            self.length_domain,
+            domain=Reals,
+            initialize=1.0,
+            doc="Particle Reynolds number",
+            units=pyunits.dimensionless,
+        )
+
+        self.Pr_particle = Var(
+            self.flowsheet().time,
+            self.length_domain,
+            domain=Reals,
+            initialize=1.0,
+            doc="Prandtl number of gas in bed",
+            units=pyunits.dimensionless,
+        )
+
+        self.Nu_particle = Var(
+            self.flowsheet().time,
+            self.length_domain,
+            domain=Reals,
+            initialize=1.0,
+            doc="Particle Nusselt number",
+            units=pyunits.dimensionless,
+        )
+        self.gas_solid_htc = Var(
```

Comment:
Would `heat_transfer_coefficient` be a sufficient name for this? We generally prefer fully descriptive names, but if we include the `gas_solid` part this would get rather long.

Reply:
From my experience working with these gas solid contactor models in the past, I believe `gas_solid_htc` is preferred to `heat_transfer_coefficient` since the former name describes the domain. Using `gas_solid_htc` is consistent with the HTC variable in the 1D moving bed model, and clarifies the physical meaning of the variable. For example, the BFB reactor has multiple HTC variables over specific domains which are abbreviated for the domain names.

Reply:
Good suggestion but I agree with @bpaul4. There could be multiple different heat transfer coefficients. For example, besides having one for gas to solid we could also have one for solid to wall or solid to heat exchanger tube if the model is expanded.